middleware-catalog-module/subject/Subject/src/main/java/co/tpg/catalog/function/SubjectFunction.java

```diff
@@ -0,0 +1,156 @@
+package co.tpg.catalog.function;
+
+import co.tpg.catalog.dao.DAO;
+import co.tpg.catalog.dao.exception.BackendException;
+import co.tpg.catalog.exception.ProcessingException;
+import co.tpg.catalog.model.Subject;
+import co.tpg.catalog.request.HttpMethod;
+import co.tpg.catalog.request.SubjectRequest;
+import co.tpg.catalog.response.AbstractResponse;
+import co.tpg.catalog.response.ErrorResponse;
+import co.tpg.catalog.response.SubjectListResponse;
+import co.tpg.catalog.response.SubjectResponse;
+import co.tpg.catalog.dao.SubjectDAO;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDB;
+import com.amazonaws.services.dynamodbv2.AmazonDynamoDBClientBuilder;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.model.ResourceNotFoundException;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+
+import javax.servlet.http.HttpServletResponse;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+
+
+/**
+ * Handler for requests to Subject.
+ * @author maxx
+ * @since 2019-10-03
+ */
+public class SubjectFunction implements RequestHandler<SubjectRequest, AbstractResponse> {
+
+    public AbstractResponse handleRequest(final SubjectRequest input, final Context context) {
+        final String DYNAMO_TABLE_NAME = "Subject";
+        final AmazonDynamoDB dynamoDB = AmazonDynamoDBClientBuilder.defaultClient();
+        final SubjectResponse response = new SubjectResponse();
+        final SubjectDAO subjectDAO = new SubjectDAO();
+        final ErrorResponse errorResponse = new ErrorResponse();
+        final Map<String, String> headers = new HashMap<>();
+        final DynamoDBMapper mapper = new DynamoDBMapper(dynamoDB);
+        String id;
+
+        headers.put("Content-Type", "application/json");
+        response.setHeaders(headers);
+
+        errorResponse.setStatusCode(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        context.getLogger().log(String.format("HTTP method: %s\n",input.getHttpMethod()));
+
+        try {
+            switch (input.getHttpMethod()) {
+                case HttpMethod.POST: {
```

Comment:
The class is not reduced with cohesion, it should follow what has been done at : https://github.com/tpg-nz/schoolncloud/blob/master/middleware-catalog-module/teaching-class/TeachingClass/src/main/java/co/tpg/catalog/function/TeachingClassFunction.java.